contracts/Genesis.sol

```diff
@@ -171,38 +230,46 @@ contract Genesis is ERC721, VRFConsumerBase, Ownable {
     }
 
     /**
-     * Verifies the proof of the sender to confirm they are in the whitelist
+     * Verifies the proof of the sender to confirm they are in given list
      * @param nonce nonce to be used
+     * @param root Merkle tree root
      * @param proof proof
      * @return valid TRUE if the proof is valid, FALSE otherwise
      */
-    function verifyProof(uint256 nonce, bytes32[] memory proof) internal view returns (bool valid) {
-        return MerkleProof.verify(proof, merkleTreeRoot, generateLeaf(nonce, msg.sender));
+    function verifyProof(uint256 nonce, bytes32 root, bytes32[] memory proof) internal view returns (bool valid) {
+        return MerkleProof.verify(proof, root, generateLeaf(nonce, msg.sender));
```

Comment:
One thing I noticed we aren't doing is checking if the `root` is actually set. What if it isn't?
Doesn't seem like it will fail, we need to test that and perhaps add a `require` here to check that the root isn't `0`?